src/Workspaces/Core/Portable/TemporaryStorage/TemporaryStorageServiceFactory.cs

```diff
@@ -364,10 +364,10 @@ private async Task WriteStreamMaybeAsync(Stream stream, bool useAsync, Cancellat
                         throw new InvalidOperationException(WorkspacesResources.Temporary_storage_cannot_be_written_more_than_once);
                     }
 
-                    if (stream.Length == 0)
-                    {
-                        throw new ArgumentOutOfRangeException();
-                    }
+                    //if (stream.Length == 0)
+                    //{
+                    //    throw new ArgumentOutOfRangeException();
+                    //}
```

Comment:
💭 Can we avoid writing to temporary storage if the string is empty?

Reply:
Good question!  i'd prefer to avoid the extra complexity for what seems like a very small savings.  

Are you ok with that?